### PR TITLE
Release v1.28.0

### PR DESCRIPTION
**Pull request by angrybayblade**
## Release summary

Version number: v1.28.0

## Release details

AEA:
- Adds checks to make sure the author name and the package name are in snake case only
- Adds tools for automating the protocol tests
- Updates the test command to spawn a process for running the pytest command to make sure there are no issues with the test coverage
- Fixes a race condition found in the `AsyncMultiplexer`

Plugins:
- Makes the fetchai ledger plugin dependent on the cosmos plugin to prevent code duplication

Tests:
- Adds a test to showcase a race condition in `AsyncMultiplexer`

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side), from `develop`
- [ ] Lint and unit tests pass locally and in CI
- [ ] I have checked the fingerprint hashes are correct by running (`aea packages lock --check`)
- [ ] I have regenerated the latest API docs
- [ ] I built the documentation and updated it with the latest changes
- [ ] I have added an item in `HISTORY.md` for this release
- [ ] I bumped the version number in the `aea/__version__.py` file.
- [ ] I bumped the version number in every Docker image of the repo and published it. Also, I built and published them with tag `latest`  
      (check the READMEs of [`aea-develop`](../develop-image/README.md#publish) 
      and [`aea-user`](../user-image/README.md#publish))
- [ ] I have pushed the latest packages to the registry.
- [ ] I have uploaded the latest `aea` to PyPI.
- [ ] I have uploaded the latest plugins to PyPI.

## Further comments

Write here any other comment about the release, if any.

### CODE REVIEW

**Review by 0xArdi — APPROVED** [review]

File: deploy-image/Dockerfile [hunk 1]
```diff
@@ -16,7 +16,7 @@ RUN apk add --no-cache go
 
 # aea installation
 RUN pip install --upgrade pip
-RUN pip install --upgrade --force-reinstall open-aea[all]==1.27.0.post1 "open-aea-cli-ipfs<2.0.0,>=1.27.0.post4"
+RUN pip install --upgrade --force-reinstall open-aea[all]==1.28.0 "open-aea-cli-ipfs<2.0.0,>=1.28.0.post4"
```
Comment (DavidMinarsch) [inline deploy-image/Dockerfile:19]: remove `.post4`

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]
I suggest you extend https://github.com/valory-xyz/open-aea/blob/main/.github/workflows/docker.yml with https://github.com/valory-xyz/open-aea/issues/513 so that we fully automate the release flow now. Specifically, steps 11-15 here: https://github.com/valory-xyz/open-aea/blob/main/scripts/RELEASE_PROCESS.md

Then also update RELEASE_PROCESS readme

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+          cd ../..
+
+
+      - name: Publish Package to PyPI
+        <<: *publish
+
+      - name: Publish cli-benchmark Plugin to PyPI
+        <<: *publish
+          <<: *pypi-config
+            packages_dir: plugins/aea-cli-benchmark/dist
```
Comment (angrybayblade) [inline .github/workflows/release.yml:79]: Ref: https://github.com/trzsz/trzsz/blob/cf36cdf29330ed930c37336abddaf3bb931c532d/.github/workflows/publish-to-pypi.yml#L45

Comment (8ball030) [reply .github/workflows/release.yml:77]: Aww you didnt reference my issue :'(

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+          pipenv --python 3.10
+
+          echo Processing open-aea
+          pipenv run make dist
+
+          echo Processing aea-cli-benchmark
+          cd plugins/aea-cli-benchmark
+          pipenv run python setup.py sdist bdist_wheel
+          cd ../..
+
```
Comment (8ball030) [inline .github/workflows/release.yml:49]: Love the alternative to the for loop

**Review by 8ball030 — APPROVED** [review]

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**